migrate-components.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+
+# Migrates the spec.containerImage field to status.lastPromotedImage if
+# status.lastPromotedImage is not already set.
+#
+# If spec.containerImage has been updated by imageRepository (has not sha)
+# then the script will attempt to get the latest push snapshot or override
+# snapshot that contains the component and gets the image from there
+#
+# REQUIREMENTS:
+#  - kubectl >= 1.24
+#  - jq
+
+# takes two snapshot jsons and returns whichever has the newest metadata.creationTimestamp
+get_newest_snapshot_json() {
+    PUSH_SNAPSHOT_JSON=$1
+    OVERRIDE_SNAPSHOT_JSON=$2
+
+    # otherwise, return whichever is younger
+    PS_TIMESTAMP=$(echo $PUSH_SNAPSHOT_JSON | jq .metadata.creationTimestamp)
+    OS_TIMESTAMP=$(echo $OVERRIDE_SNAPSHOT_JSON | jq .metadata.creationTimestamp)
+    if [[ $OS_TIMESTAMP > $PS_TIMESTAMP ]]; then
```

Comment:
```suggestion
    if [[ "$OS_TIMESTAMP" > "$PS_TIMESTAMP" ]]; then
```

Reply:
jq already outputs those values as strings here, I  tested this pretty thoroughly and it works as expected, so I don't want to try quoting the variables in cases that changes the behavior (it shouldn't, but I don't really trust bash)